internal/resources/grafana/resource_sso_settings_test.go

```diff
@@ -0,0 +1,292 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/grafana/grafana-openapi-client-go/client"
+	"github.com/grafana/grafana-openapi-client-go/models"
+
+	"github.com/grafana/terraform-provider-grafana/internal/common"
+	"github.com/grafana/terraform-provider-grafana/internal/testutils"
+)
+
+func TestSSOSettings_basic(t *testing.T) {
+	testutils.CheckOSSTestsEnabled(t, ">=10.4.0")
+
+	providers := []string{"github", "gitlab", "google", "generic_oauth", "azuread", "okta"}
+
+	api := testutils.Provider.Meta().(*common.Client).GrafanaOAPI.WithOrgID(1)
+
+	for _, provider := range providers {
+		defaultSettings, err := api.SsoSettings.GetProviderSettings(provider)
+		if err != nil {
+			t.Fatalf("failed to fetch the default settings for provider %s: %v", provider, err)
+		}
+
+		resourceName := fmt.Sprintf("grafana_sso_settings.%s_sso_settings", provider)
+
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: testutils.ProviderFactories,
+			CheckDestroy:      checkSsoSettingsReset(api, provider, defaultSettings.Payload),
+			Steps: []resource.TestStep{
+				{
+					Config: testConfigForProvider(provider, "new"),
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr(resourceName, "provider_name", provider),
+						resource.TestCheckResourceAttr(resourceName, "oauth2_settings.#", "1"),
+						resource.TestCheckResourceAttr(resourceName, "oauth2_settings.0.client_id", fmt.Sprintf("new_%s_client_id", provider)),
+						resource.TestCheckResourceAttr(resourceName, "oauth2_settings.0.client_secret", fmt.Sprintf("new_%s_client_secret", provider)),
+					),
+				},
+				{
+					Config: testConfigForProvider(provider, "updated"),
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr(resourceName, "provider_name", provider),
+						resource.TestCheckResourceAttr(resourceName, "oauth2_settings.#", "1"),
+						resource.TestCheckResourceAttr(resourceName, "oauth2_settings.0.client_id", fmt.Sprintf("updated_%s_client_id", provider)),
+						resource.TestCheckResourceAttr(resourceName, "oauth2_settings.0.client_secret", fmt.Sprintf("updated_%s_client_secret", provider)),
+					),
+				},
+			},
+		})
+	}
+}
+
+func TestSSOSettings_applyExample(t *testing.T) {
+	testutils.CheckOSSTestsEnabled(t, ">=10.4.0")
+
+	var settings models.GetProviderSettingsOKBody
+
+	provider := "github"
+
+	api := testutils.Provider.Meta().(*common.Client).GrafanaOAPI.WithOrgID(1)
+	defaultSettings, err := api.SsoSettings.GetProviderSettings(provider)
+	if err != nil {
+		t.Fatalf("failed to fetch the default settings for provider %s: %v", provider, err)
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: testutils.ProviderFactories,
+		CheckDestroy:      checkSsoSettingsReset(api, provider, defaultSettings.Payload),
+		Steps: []resource.TestStep{
+			{
+				Config: testutils.TestAccExample(t, "resources/grafana_sso_settings/resource.tf"),
```

Comment:
Why skip the example if we're running it here?

Reply:
If you refer to the changes from `internal/resources/examples_test.go` then I'm skipping the example there because the Grafana API for the sso settings is not yet available and the example will fail.

Reply:
Instead of a skip in the example, you could put the version condition and remove the test in this file